src/server/pfs/server/master.go

```diff
@@ -96,7 +123,21 @@ func (d *driver) finishCommits(ctx context.Context) error {
 		repos[key] = cancel
 		go func() {
 			backoff.RetryUntilCancel(ctx, func() error { //nolint:errcheck
-				return d.finishRepoCommits(ctx, compactor, key)
+				lockCtx, err := ring.Lock(ctx, lockPrefix)
+				if err != nil {
+					return errors.Wrap(err, "error locking repo lock")
+				}
+				defer func() {
+					unlockErr := ring.Unlock(lockPrefix)
+					if err == nil {
```

Comment:
This isn't going to do what you expect. You would need a named return to check whether `finishRepoCommits` returned an error. The common pattern for this is `retErr` being the named return and checking that in the defer. We may want to take a look at the `multierr` package for this so we can capture the `finishRepoCommits` error and `Unlock` error.